\documentclass[10pt]{article}
\usepackage{tgpagella}
\usepackage{xcolor,graphicx}
\usepackage{hyperref}
\usepackage{minted}
\newminted[shellcode]{shell}{bgcolor=lightgray}

%\setmainfont[Ligatures=TeX]{TeX Gyre Pagella}

\begin{document}

\begin{titlepage}
\vfill
{\Huge\centering%
TRUmiCount\\
How-To\\}
\vskip5cm
{\large\centering%
Florian G. Pflug\\
\texttt{<florian.pflug@univie.ac.at>}\\}
\vfill
\end{titlepage}

\section{Installing TRUmiCount}

\subsection{Installation via Conda (Recommended)}

\subsubsection*{Installing Conda}

Conda is a package manager that allows easy installation of a large range of software packages. See \url{https://conda.io/docs/user-guide/install/index.html} for your options of how to insteall conda. Briefly, on 64-bit linux do\footnote{Instead of \texttt{/conda}, you can choose any other directory to install conda into}

\begin{shellcode}
INSTALLER=Miniconda2-latest-Linux-x86_64.sh
CONDA_DIR=/conda
curl -O https://repo.continuum.io/miniconda/$INSTALLER
bash $INSTALLER -p $CONDA_DIR
\end{shellcode}

\subsubsection*{Creating an environment}

Conda allows the creation of multiple \emph{environments}, each containing different collections of packages. We will now create an environment for TRumiCount

\begin{shellcode}
$CONDA_DIR/bin/conda create -n trc
\end{shellcode}

This environment is now \emph{activated} to make it the target of further conda commands, and the installed software visible. This must be done every time a new terminal window is opened!

\begin{shellcode}
source $CONDA_DIR/bin/activate trc
\end{shellcode}
%$
\subsubsection*{Installing BioConda}

Conda packages are organized into so-called \emph{channels}. We add the BioConda channel which provides many common tools for dealing with high-throughput sequencing data

\begin{shellcode}
conda config --env --add channels defaults
conda config --env --add channels conda-forge
conda config --env --add channels bioconda
\end{shellcode}

\subsubsection*{Installing TRumiCount}

Finall we add the channel that supplies TRumiCount and a modified version of umi\_tools with improved handling of paired-end reads\footnote{Note that the backslash (``\textbackslash'') only serves to make your shell ignore the linebreak that follows it. If you enter the command as a single line, skip the backslash}

\begin{shellcode}
conda config --env --add channels \
  http://tuc:tuc@www.cibiv.at/~pflug_/conda.trumicount/
\end{shellcode}

TRumiCount and our version of umi\_tools can now be installed

\begin{shellcode}
conda install trumicount umi_tools
\end{shellcode}

\section{Using TRUmiCount}

\subsection{Supported Input Formats \& Options}

To be able to separate true UMIs from biases and to estimate the percentage of true UMIs that are lost during library preparation or data processing, TRUmiCount analyses the distribution of read counts per UMI for each gene (or any other type of genomic feature).

\subsubsection*{Reading BAM Files}

If an BAM File is provided as input with ``\texttt{--input-bam BAMFILE}'', TRUmiCount uses umi\_tools's \texttt{group} tool to extract a list of error-corrected UMIs and their read counts. By default, it is assumed that
\begin{itemize}
  \item The sequence name corresponds to the gene name. Alternatively, the gene names can be stored in BAM file tags -- this can be changed by using umi\_tool's \texttt{--gene-tag=GENE\_TAG} option. To tell TRUmiCount to invoke umi\_tools using that option, use ``\texttt{--umitools-option --gene-tag=\allowbreak GENE\_TAG}'' when invoking TRUmiCount.
  \item The UMI was appended to the read name, and separated by ``:''. A different separator can be specified with ``\texttt{--umi-sep SEPARATOR}''
  \item The BAM file contains single-end reads (read2 is ignored). To take the mapping position of both mates into account when grouping reads by UMI, specify \texttt{--paired}.
  \item Reads with a mapping quality below 20 should be ignored. This threshold can be changed with ``\texttt{--mapping-quality MAPQ}''.
  \item Each UMI initially had two copies. This is e.g. the case of molecules before amplification were double-stranded and the copies produced from both strands are identical. This number can be changed with ``\texttt{--molecules COPIES}''
  \item UMIs must be supported by at least two reads to be assumed to be a true UMI and not a phantom. This threshold can be changed with ``\texttt{--threshold TH}''.
  \item The BAM file must have a corresponding index. A suitable index can be created with ``\texttt{samtools index BAMFILE}''.
\end{itemize}

\subsubsection*{Reading grouped UMIs}

XXX Write Me

\subsection{Examples}

\subsubsection*{Downloading Example Data}

XXX Write Me (plus conversion from CRAM to BAM)

\subsubsection*{Single-End Data}

\begin{shellcode}
samtools index kv_100g.bam
trumicount --input-bam kv_100g.bam \
  --molecules 2 --threshold 5 \
  --output-plot kv_100g.pdf --plot-x-bin 10 \
  --output-counts kv_100g.tab
  --cores 4
\end{shellcode}

\subsection{Paired-End Data}

\begin{shellcode}
samtools index sg_100g.bam
trumicount --input-bam sg_100g.bam --umipair-sep '-'\
  --paired --filter-strand-umis --molecules 1 --threshold 24 \
  --output-plot sg_100g.pdf --plot-x-bin 3 \
  --output-counts sg_100g.tab
  --cores 4
\end{shellcode}

{\centering \includegraphics[width=7cm]{examples/sg_100g.pdf}\\}

\end{document}
