\documentclass[10pt]{article}
%\usepackage{tgpagella}
\usepackage{mathpazo}
\usepackage{xcolor,graphicx}
\usepackage{hyperref}
\usepackage{minted}
\newminted[shellcode]{shell}{bgcolor=lightgray}

%\setmainfont[Ligatures=TeX]{TeX Gyre Pagella}

\begin{document}

\begin{titlepage}
\vfill
{\Huge\centering%
TRUmiCount\\
Manual \& How-To\\}
\vskip5cm
{\large\centering%
Florian G. Pflug\\
\texttt{<florian.pflug@univie.ac.at>}\\}
\vfill
\end{titlepage}

\section{Installing TRUmiCount}

\subsection{Installation via Conda (Recommended)}

\subsubsection*{Installing Conda}

Conda is a package manager that allows easy installation of a large range of software packages. See \url{https://conda.io/docs/user-guide/install/index.html} for your options of how to insteall conda. Briefly, on 64-bit linux do\footnote{Instead of \texttt{/conda}, you can choose any other directory to install conda into}

\begin{shellcode}
INSTALLER=Miniconda2-latest-Linux-x86_64.sh
CONDA_DIR=/conda
curl -O https://repo.continuum.io/miniconda/$INSTALLER
bash $INSTALLER -p $CONDA_DIR
\end{shellcode}

\subsubsection*{Creating an environment}

Conda allows the creation of multiple \emph{environments}, each containing different collections of packages. We will now create an environment for TRUmiCount

\begin{shellcode}
$CONDA_DIR/bin/conda create -n trc
\end{shellcode}

This environment is now \emph{activated} to make it the target of further conda commands, and the installed software visible. This must be done every time a new terminal window is opened!

\begin{shellcode}
source $CONDA_DIR/bin/activate trc
\end{shellcode}
%$
\subsubsection*{Installing BioConda}

Conda packages are organized into so-called \emph{channels}. We add the BioConda channel which provides many common tools for dealing with high-throughput sequencing data

\begin{shellcode}
conda config --env --add channels defaults
conda config --env --add channels conda-forge
conda config --env --add channels bioconda
\end{shellcode}

\subsubsection*{Installing TRUmiCount}

Finall we add the channel that supplies TRUmiCount and a modified version of umi\_tools with improved handling of paired-end reads\footnote{Note that the backslash (``\textbackslash'') only serves to make your shell ignore the linebreak that follows it. If you enter the command as a single line, skip the backslash}

\begin{shellcode}
conda config --env --add channels \
  http://tuc:tuc@www.cibiv.at/~pflug_/conda.TRUmiCount/
\end{shellcode}

TRUmiCount and our version of umi\_tools can now be installed

\begin{shellcode}
conda install TRUmiCount umi_tools samtools
\end{shellcode}

\section{Using TRUmiCount}

\subsection{Supported Input Formats \& Options}

To be able to separate true UMIs from biases and to estimate the percentage of true UMIs that are lost during library preparation or data processing, TRUmiCount analyses the distribution of read counts per UMI for each gene (or any other type of genomic feature).

\subsubsection*{Reading BAM Files}

If an BAM File is provided as input with ``\texttt{--input-bam BAMFILE}'', TRUmiCount uses umi\_tools's \texttt{group} tool to extract a list of error-corrected UMIs and their read counts. By default, it is assumed that
\begin{itemize}
  \item The sequence name corresponds to the gene name. Alternatively, the gene names can be stored in BAM file tags -- this can be changed by using umi\_tool's \texttt{--gene-tag=GENE\_TAG} option. To tell TRUmiCount to invoke umi\_tools using that option, use ``\texttt{--umitools-option --gene-tag=\allowbreak GENE\_TAG}'' when invoking TRUmiCount.
  \item The UMI was appended to the read name, and separated by ``:''. A different separator can be specified with ``\texttt{--umi-sep SEPARATOR}''
  \item The BAM file contains single-end reads (read2 is ignored). To take the mapping position of both mates into account when grouping reads by UMI, specify \texttt{--paired}.
  \item Reads with a mapping quality below 20 should be ignored. This threshold can be changed with ``\texttt{--mapping-quality MAPQ}''.
  \item Each UMI initially had two copies. This is e.g. the case of molecules before amplification were double-stranded and the copies produced from both strands are identical. This number can be changed with ``\texttt{--molecules COPIES}''
  \item UMIs must be supported by at least two reads to be assumed to be a true UMI and not a phantom. This threshold can be changed with ``\texttt{--threshold TH}''.
  \item The BAM file must have a corresponding index. A suitable index can be created with ``\texttt{samtools index BAMFILE}''.
\end{itemize}

\subsubsection*{Reading grouped UMIs}

XXX Write Me

\subsection{Output}

XXX Write Me

\section{Examples}

Explain where to find the example data

\subsection*{Single-End Data}

The file \texttt{kv\_1000g.bam} contains a reduced (restricted to the first 0100 genes, and subsamples to 50\%) version of data published by Kivioja \textit{et al.}\footnote{Kivioja, T. \textit{et al.} Counting absolute numbers of molecules using unique molecular identifiers. \textit{Nature Methods} \textbf{9}, 72â€“74 (2011)}.

\begin{shellcode}
samtools index kv_1000g.bam
trumicount --input-bam kv_1000g.bam \
  --molecules 2 --threshold 2 \
  --output-plot kv_1000g.pdf --plot-x-bin 1 \
  --output-counts kv_1000g.tab \
  --cores 4
\end{shellcode}

This produces the following diagnostic plot \texttt{kv\_1000g.pdf} showing the observed distribution of reads per UMI and our model's predicted distribution and loss.

{\centering \includegraphics[width=7cm]{../examples/kv_1000g.pdf}\\}

\subsection{Paired-End Data with Stranded UMIs}

The file \texttt{sg\_100g.bam} contains a reduced (restricted to the first 100 genes, and subsamples to 25\%) version of data published by Shiroguchi \textit{et al.}\footnote{Shiroguchi, K., Jia, T. Z., Sims, P. A. \& Xie, X. S. Digital RNA sequencing minimizes sequence-dependent bias and amplification noise with optimized single-molecule barcodes. \textit{Proceedings of the National Academy of Sciences of the United States of America} \textbf{109}, 1347-1352 (2012).} and was produced by ligating Y-shaped (i.e. not fully complementary) adapters to the cDNA fragments. This makes copies produced from the two strands of these ligation products distinguishable, meaning each original molecule gives to two UMIs. 

\subsubsection*{Option 1 -- Filtering out incomplete strand UMIs}

The data from from Shiroguchi \textit{et al.} is analyzed in paired-end mode, and UMIs are filtered-out if the UMI corresponding to the second strand of the original template molecule is not detected.

Note that the actual loss rate in this mode is not simply the probability $\mathbb{P}(C < T)$ of an UMI having fewer than $T$ reads. Since we filter out UMIs \emph{either} if they themselves have fewer than $T$ reads, \emph{or} if their partner UMI wasn't detected at all, the actual loss is $1 - (1 - \mathbb{P}(C < T))\cdot (1 - \mathbb{P}(C=0))$. TRUmiCount adjust the loss computation accordingly and outputs corrected losses. The model distribution shown in the plots, however,  does not (and cannot) take this adjustment into account, so that the stated loss is no longer simply the sum of the model probabilities for read counts less than $T$.

\begin{shellcode}
samtools index sg_100g.bam
trumicount --input-bam sg_100g.bam --umipair-sep '-'\
  --paired --filter-strand-umis --molecules 1 --threshold 24 \
  --output-plot sg_100g.pdf --plot-x-bin 3 \
  --output-counts sg_100g.tab \
  --cores 4
\end{shellcode}

This produces the following diagnostic plot \texttt{sg\_100g.pdf} showing the observed distribution of reads per UMI and our model's predicted distribution and loss.

{\centering \includegraphics[width=7cm]{../examples/sg_100g.pdf}\\}

\subsubsection*{Option 2 -- Combining strand UMIs}

XXX: Is still still broken, and how?

The data from from Shiroguchi \textit{et al.} is analyzed in paired-end mode, this time combining UMIs stemming from the two strands of a single template molecule. UMIs without a mate to combine with are dropped (as they are for ``\texttt{--filter-strand-umis}''.

Note that the actual loss is again not simply $\mathbb{P}(C < T)$, but in this case is $1 - (1-\mathbb{P}(C < T))^2$, because we now filter out UMI pairs if \emph{either} of the partners have fewer than $T$ reads. This is taken into account by TRUmiCount to compute the loss, and the stated loss there again isn't simply the sum of the model probabilities for read counts less than $T$. 

\begin{shellcode}
samtools index sg_100g.bam
trumicount --input-bam sg_100g.bam --umipair-sep '-'\
  --paired --combine-strand-umis --molecules 1 --threshold 24 \
  --output-plot sg_100g_comb.pdf --plot-x-bin 3 \
  --output-counts sg_100g_comb.tab \
  --cores 4
\end{shellcode}

This produces the following diagnostic plot \texttt{sg\_100g\_comb.pdf} showing the observed distribution of reads per UMI and our model's predicted distribution and loss.

{\centering \includegraphics[width=7cm]{../examples/sg_100g_comb.pdf}\\}

\end{document}
