<html>
<head>
  <meta charset='utf-8' />
  <meta name="description" content="TRUmiCount: Correctly counting molecules using unique molecular identifiers" />
  <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">
  <title>TRUmiCount</title>
</head>
<body>

<!-- HEADER -->
<div id="header_wrap" class="outer">
    <header class="inner">
      <a id="forkme_banner" href="https://github.com/Cibiv/trumicount">View on GitHub</a>
      <a id="manual_banner" href="manual.pdf">Manual</a>

      <h1 id="project_title">TRUmiCount</h1>
      <h2 id="project_tagline">Correctly counting molecules using unique molecular identifiers</h2>
    </header>
</div>

<!-- MAIN CONTENT -->
<div id="main_content_wrap" class="outer">
<section id="main_content" class="inner">

<h2>Description</h2>
<strong>Motivation:</strong>
Counting molecules using <i>next-generation sequencing</i> (NGS) suffers from
PCR amplification bias, which reduces the accuracy of many quantitative
NGS-based experimental methods such as RNA-Seq. This is true even if molecules
are made distinguishable using <i>unique molecular identifiers</i> (UMIs) before
PCR amplification, and distinct UMIs are counted instead of reads: Molecules
that are lost entirely during the sequencing process will still cause
under-estimation of the molecule count, and amplification artifacts like PCR
chimeras create phantom UMIs and thus cause over-estimation.<br>

<strong>Results:</strong>
TRUmiCount algorithm corrects for both types of errors. It is based on a
mechanistic model of PCR amplification and sequencing, whose two parameters
have an immediate physical interpretation as PCR efficiency and sequencing
depth and can be estimated from experimental data without requiring calibration
experiments or spike-ins. We show that our model captures the main stochastic
properties of amplification and sequencing, and that it allows us to filter out
phantom UMIs and to estimate the number of molecules lost during the sequencing
process. Finally, we demonstrate that the phantom-filtered and loss-corrected
molecule counts computed by TRUmiCount measure the true number of molecules with
considerably higher accuracy than the raw number of distinct UMIs.

<h2>Availability</h2>
<h3 a=conda>Conda Repository (also includes gwpcR)</h3>
<b>Repository Url:</b> <var>http://www.cibiv.at/~pflug_/conda/</var><br>
<b>Available Packages:</b> <var>trumicount</var>, <var>umi_tools</var> (patched
version, see <a href="https://github.com/CGATOxford/UMI-tools">here</a> for
the original)<br>
If you already have <a href="https://bioconda.github.io">bioconda</a> or plain
<a href="https://conda.io">conda</a> installed, TRUmiCount is installed into a
separate conda environment <it>trc</it> with
<pre>
conda create -n trc
source activate trc
conda config --env --add channels defaults
conda config --env --add channels conda-forge
conda config --env --add channels bioconda
conda config --env --add channels http://www.cibiv.at/~pflug_/conda/
conda install trumicount umi_tools samtools
</pre>
For more detailed instruction and other installation options see the <a
href="manual.pdf">manual</a>.
<h3>Soure Code</h3>
<a class="github_link" href="https://github.com/Cibiv/trumicount.git">View on GitHub</a>
<h2>Example Data</h2>
<a href="kv_1000g.bam"><b>kv_1000g.bam</b></a>:
Mapped single-end reads for first 1000 genes from replicate 1 of the D.
melanogaster data by Kivioja <i>et al.</i> (Counting absolute numbers of
molecules using unique molecular identifiers. <i>Nature Methods</i> <b>9</b>,
72-74, 2011)
<br>
<a href="sg_100g.bam"><b>sh_100g.bam</b></a>:
Mapped reads for first 100 genes from replicate 1 of the E. coli data by
Shiroguchi <i>et al.</i> (Shiroguchi <i>et al.</i> Digital RNA sequencing
minimizes sequence-dependent bias and amplification noise with optimized
single-molecule barcodes. <i>PNAS</i> <b>109</b>, 1347-1352, 2012)

<!-- END MAIN CONTENT -->
</section>
</div>

<!-- FOOTER  -->
<div id="footer_wrap" class="outer">
  <footer class="inner">
    <p class="copyright">TRUmiCount maintained by the <a href="https://github.com/Cibiv">
    Center for Integrative Bioinformatics Vienna</a></p>
    <p>Published with <a href="http://pages.github.com">GitHub Pages</a></p>
  </footer>
</div>

</body>
