<html>
<head>
  <meta charset='utf-8' />
  <meta name="description" content="TRUmiCount: Correctly counting molecules using unique molecular identifiers" />
  <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">
  <title>TRUmiCount</title>
</head>
<body>

<!-- HEADER -->
<div id="header_wrap" class="outer">
    <header class="inner">
      <div id="top">
        <a id="manual_banner" class="top_banner" href="manual.pdf">Manual</a>
        <a id="publication_banner" class="top_banner"  href="https://doi.org/10.1093/bioinformatics/bty283">Publication</a>
        <a id="forkme_banner" class="top_banner"  href="https://github.com/Cibiv/trumicount">View on GitHub</a>
      </div>
      <div id="bottom" class="bottom">
        <a class="bottom_banner" href="https://anaconda.org/bioconda/trumicount">
          <img src='https://anaconda.org/bioconda/trumicount/badges/installer/conda.svg' class=inlineimg style='height: auto'>
        </a>
        <a class="bottom_banner" href="https://anaconda.org/bioconda/trumicount">
          <img src='https://anaconda.org/bioconda/trumicount/badges/platforms.svg' class=inlineimg style='height: auto'>
        </a>
        <a class="bottom_banner" href="https://anaconda.org/bioconda/trumicount">
          <img src='https://anaconda.org/bioconda/trumicount/badges/version.svg' class=inlineimg style='height: auto'>
        </a>
        <a class="bottom_banner" href='#license'>
          <img src='https://anaconda.org/bioconda/trumicount/badges/license.svg' class=inlineimg style='height: auto'>
        </a>
      </div>

      <h1 id="project_title">TRUmiCount</h1>
      <h2 id="project_tagline">Correctly counting molecules using unique molecular identifiers</h2>
    </header>
</div>

<!-- MAIN CONTENT -->
<div id="main_content_wrap" class="outer">
<section id="main_content" class="inner">

<h2>Description</h2>
<strong>Motivation:</strong>
Counting molecules using <i>next-generation sequencing</i> (NGS) suffers from
PCR amplification bias, which reduces the accuracy of many quantitative
NGS-based experimental methods such as RNA-Seq. This is true even if molecules
are made distinguishable using <i>unique molecular identifiers</i> (UMIs) before
PCR amplification, and distinct UMIs are counted instead of reads: Molecules
that are lost entirely during the sequencing process will still cause
under-estimation of the molecule count, and amplification artifacts like PCR
chimeras create phantom UMIs and thus cause over-estimation.<br>

<p>
<strong>Results:</strong>
TRUmiCount uses a mechanistic model of PCR amplification to correct for both
types of errors. In our <a href="https://doi.org/10.1093/bioinformatics/bty283">
paper</a> we demonstrate that the phantom-filtered and loss-corrected molecule counts
computed by TRUmiCount measure the true number of molecules with considerably
higher accuracy than the raw number of distinct UMIs.<br>

<h2>Publication</h2>
TRUmiCount is described in detail in our paper:

<p>Florian G. Pflug, Arndt von Haeseler. (2018). TRUmiCount: Correctly counting
absolute numbers of molecules using unique molecular identifiers.
<i>Bioinformatics</i>, DOI: <a href="https://doi.org/10.1093/bioinformatics/bty283">
10.1093/bioinformatics/bty283</a>.

<p>If you use TRUmiCount, please cite this publication!

<h2>Availability</h2>

<h3 a=conda>Install from <a href='https://bioconda.github.io/'><img src='bioconda.png'
class="inlineimg"></a></h3>
If you're already using <a href="https://bioconda.github.io">Conda</a>, you can
install TRUmiCount from the <a href="https://conda.io">Bioconda</a>
channel by doing
<pre>
conda install -c bioconda trumicount samtools
</pre>

<p>
TRUmiCount relies on <a href="http://github.com/CGATOxford/UMI-tools">UMI-Tools</a>
to extract the UMIs associated with a gene. We provide a modified version that is
more seamlessly integrated with TRUmiCount which you can install with
<pre>
conda install -c cibiv umi_tools_tuc
</pre>
(You can also use the umodified UMI-Tools with TRUmiCount, but for single-cell
data in particular you might encounter some issues)

<p>
For more detailed instructions and other installation options see the
<a href="manual.pdf">manual</a>.

<h3>Sourecode on <img src='images/github.png' class='inlineimg' style='height: 2.2ex; bottom: -0.4ex'></h3>
The sourcecode of TRUmiCount is avaiable on <a href="https://github.com/Cibiv/trumicount.git">
https://github.com/Cibiv/trumicount.git</a>. Our modified version of UMI-Tools is available at
<a href="https://github.com/Cibiv/UMI-Tools/tree/tuc-patches">https://github.com/Cibiv/UMI-Tools/tree/tuc-patches</a>

<h2>Usage Examples</h2>
<h3>Datasets</h3>
<a href="kv_1000g.bam"><b><var>kv_1000g.bam</var></b></a>:
Mapped single-end reads for first 1000 genes from replicate 1 of the D.
melanogaster data by Kivioja <i>et al.</i> (Counting absolute numbers of
molecules using unique molecular identifiers. <i>Nature Methods</i> <b>9</b>,
72-74, 2011)

<p>
<a href="sg_100g.bam"><b><var>sh_100g.bam</var></b></a>:
Mapped reads for first 100 genes from replicate 1 of the E. coli data by
Shiroguchi <i>et al.</i> (Shiroguchi <i>et al.</i> Digital RNA sequencing
minimizes sequence-dependent bias and amplification noise with optimized
single-molecule barcodes. <i>PNAS</i> <b>109</b>, 1347-1352, 2012)

<p>
<a href="10xn9k_10c.bam"><b><var>10xn9k_10c.bam</var></b></a>:
Mapped reads from 10 single mouse brain cells extracted from the
<a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.1.0/neuron_9k">
neuron_9k</a> single-cell RNA-seq example dataset provided by
<a href="https://www.10xgenomics.com/">10x Genomics</a>. The full dataset contains
about 300,000,000 reads from 9,000 individual E18 mouse brain cells.

<h3>Running TRUmiCount</h3>
To run TRUmiCount on the example data file <var>kv_1000g.bam</var>, the
firs must first be downloaded, and index with <var>samtools</var> by doing:
<pre>
curl -O https://cibiv.github.io/trumicount/kv_1000g.bam
samtools index kv_1000g.bam
</pre>

<p>
This indexed BAM-File can then be processed with trumicount to produce a
table containing bias-corrected numbers of transcript molecules for each
gene (<var>kv_1000g.tab</var>)
<pre>
trumicount --input-bam kv_1000g.bam --umi-sep ':' --molecules 2 --threshold 2 --genewise-min-umis 3 --output-counts kv_1000g.tab
</pre>

<p>
For a brief list of command-line options of TRUmiCount see <code>trumicount
--help</code>, for an in-depth description of the possible operating modes,
input and output formats and command-line options see the <a
href="manual.pdf">manual</a>

<h2><a name='license'></a>License</h2>
TRUmiCount is free software: you can redistribute it and/or modify it under
the terms of the GNU Affero General Public License as published by the Free
Software Foundation, either version 3 of the License, or (at your option) any
later version.

<p>
TRUmiCount is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
details.

<!-- END MAIN CONTENT -->
</section>
</div>

<!-- FOOTER  -->
<div id="footer_wrap" class="outer">
  <footer class="inner">
    <p class="copyright">TRUmiCount maintained by the <a href="http://www.cibiv.at">
    Center for Integrative Bioinformatics Vienna</a></p>
    <p>Published with <a href="http://pages.github.com">GitHub Pages</a></p>
  </footer>
</div>

</body>
