<html>
<body>
<h1>TRUmiCount</h1>
<h2>Availability</h2>
<h3 a=conda>Conda Repository (also includes gwpcR)</h3>
<b>Repository Url:</b> <tt>http://www.cibiv.at/~pflug_/conda/</tt><br>
<b>Available Packages:</b> <tt>trumicount</tt>, <tt>umi_tools</tt>, <tt>r-gwpcr</tt><br>
If you already have <a href="">conda</a> installed, TRUmiCount is installed
into a separate conda environment <it>trc</it> with
<pre>
conda create -n trc
source activate trc
conda config --env --add channels defaults
conda config --env --add channels conda-forge
conda config --env --add channels bioconda
conda config --env --add channels http://www.cibiv.at/~pflug_/conda/
conda install trumicount umi_tools samtools
</pre>
For more detailed instruction and other installation options see the <a
href="manual.pdf">manual</a>.
<h3>Soure Code</h3>
<b>Git Repository Url:</b> <tt>https://github.com/Cibiv/trumicount.git</tt><br>
<h2>Manual</h2>
<a href="manual.pdf">TRUmiCount Manual</a>
<h2>Example Data</h2>
<a href="kv_1000g.bam"><b>kv_1000g.bam</b></a>:
Mapped single-end reads for first 1000 genes from replicate 1 of the D.
melanogaster data by Kivioja <i>et al.</i> (Counting absolute numbers of
molecules using unique molecular identifiers. <i>Nature Methods</i> <b>9</b>,
72-74, 2011)
<br>
<a href="sg_100g.bam"><b>sh_100g.bam</b></a>:
Mapped reads for first 100 genes from replicate 1 of the E. coli data by
Shiroguchi <i>et al.</i> (Shiroguchi <i>et al.</i> Digital RNA sequencing
minimizes sequence-dependent bias and amplification noise with optimized
single-molecule barcodes. <i>PNAS</i> <b>109</b>, 1347-1352, 2012)

<h1>gwpcR (R package)<h1>
<h2>Availability</h2>
<h3>Conda Repository</h3>
The conda repository for TRUmiCount (see above) also contains a package for
gwpcR called <tt>r-gwpcr</tt>, see <a href="#conda">above</a>.
<h3>Soure Code</h3>
<b>Git Repository Url:</b> <tt>https://github.com/Cibiv/gwpcR.git</tt><br>
With the <tt>devtools</tt> package, the R package <tt>gwpcR</tt> can be installed
directly from this source code repository with the following commands (to be
executed within R on the command-line or within RStudio)
<pre>
install.packages("devtools")
library(devtools)
devtools::install_github('Cibiv/gwpcR', ref='latest-release')
</pre>
</body>
